storage/src/adb/any/variable/mod.rs

```diff
@@ -767,16 +823,21 @@ impl<E: RStorage + Clock + Metrics, K: Array, V: Codec, H: CHasher, T: Translato
             self.locations.sync().map_err(Error::Journal),
         )?;
 
-        // Prune the log up to the section containing the requested pruning location. We always
-        // prune the log first, and then prune the MMR+locations structures based on the log's
-        // actual pruning boundary. This procedure ensures all log operations always have
-        // corresponding MMR & location entries, even in the event of failures, with no need for
-        // special recovery.
+        // Prune the log up to the section containing the requested pruning location.
+        // First, we write the oldest retained location to metadata before pruning the log.
```

Comment:
might clarify you're righting the "new" oldest retained location we expect to have after pruning.